.github/workflows/push.yml

```diff
@@ -40,7 +40,7 @@ jobs:
           name: push.zip
           path: push.zip
       - name: Deploy Kdoc to github pages
-        uses: JamesIves/github-pages-deploy-action@4
+        uses: JamesIves/github-pages-deploy-action@4.2.5
         with:
```

Comment:
Sorry for the mistake, it might be better to use `@v4`?

Reply:
Woops indeed  a `v` is needed here, right? Sorry I'm not super familiar with Github actions, does`v4` mean use the latest from 4.x.y?

Reply:
yep

Reply:
That seems a recipe for the kind of supply chain attacks that the npm ecosystem is struggling with at the moment? I'd rather keep the version fixed even if it means updating it manually. At least  we keep some control...

It won't solve anything because Github will certainly allow tags to be re-created (which is a bit scary now that I think about it...). But at least it won't make it any worse?

Reply:
Sounds good to me 👍

Reply:
Actually, we should pin versions to their sha1 (see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). 

@BoD want to take a look as part of this PR?

Reply:
Sure! I'll have a 👀.

Reply:
Used sha1 for `gradle` and `JamesIves` and kept the tag names for `actions` as I figured we can trust that? LMK what you think.

Reply:
Works for me. If Github goes rogue, I guess we'll have bigger problems...